src/App.tsx

```diff
@@ -2,108 +1,75 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { ItemPerPage } from './types/ItemPerPage';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+const itemsCount: ItemPerPage[] = [3, 5, 10, 20];
+
 export const App: React.FC = () => {
+  const [selectedPage, setSelectedPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState<ItemPerPage>(5);
+
+  const startItem = itemsPerPage * selectedPage - itemsPerPage;
+  const finishItem =
+    itemsPerPage * selectedPage > items.length
+      ? items.length
+      : itemsPerPage * selectedPage;
+
+  const preparedItem = items.slice(startItem, finishItem);
+
+  const handleSetItemsPerPage = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setItemsPerPage(+event.target.value as ItemPerPage);
```

Comment:
Casting the value to 'ItemPerPage' after using the unary plus operator to convert the string to a number is redundant if your type 'ItemPerPage' is a number type. If 'ItemPerPage' is an enum or a union of specific numbers, ensure that the values from the event can actually be one of those specific numbers.